content/concepts/address-discovery-and-advertisement.md

```diff
@@ -0,0 +1,216 @@
+  `/libp2p/relay` namespace on the DHT and then randomly picking one Relay server that’s dialable.
+
+    ##### Creating long lived connections with the Relay server and advertising relay addresses
+
+    - Once a peer knows it has private reachability by using AutoNAT, the only way it can be dialled from
+    outside is through Relay servers.
+
+    - So, when a peer receivers an `EvtLocalReachabilityChanged` event with Private reachability, it searches the
+    DHT for Relay servers as described above and connects to one of them.
+
+    - It then creates a Relay address for itself by combining the address of the Relay server with it’s peerID and
+    adds it to the dialable address set of the peer. In this case, the peer also removes all public addresses from
+    it’s set of dialable addresses as it overwrites them with the Relay addresses since it does NOT have public
+    reachability. This means it will remove public interface addresses, UPnP addresses and even public observed
+    addresses it is confident about from it’s set of dialable addresses. However it will retain all
+    private IP addresses it discovers using all these mechanisms so peers on the same network can dial it
+    without using Relays.
+
+    - Note that the connection with the Relay server should be a long lived one as peers can ONLY connect to it
+    using the relay address if it is connected to the Relay server.
+
+    - If a peer discovers it has public reachability by seeing an “EvtLocalReachabilityChanged” event for it,
+    it stops advertising Relay addresses.
+
+    - When a peer with private reachability loses connection to it’s Relay server, it will remove the corresponding
+    relay addresses from it’s address set and look for a new Relay server to connect to.
+
+    - When a peer connects to a new Relay server, it adds the corresponding Relay address to it’s set
+    of dialable addresses.
+
+    - The component that listens to NAT reachability events, connects to Relay servers and builds/advertises
+    Relay addresses is called [AutoRelay][repo_autorelay]. It can be enabled by using the `libp2p.EnableRelay(nil)` and `libp2p.EnableAutoRelay()` option on the Host.
+
+{{% notice "note" %}}
+We can also configure AutoRelay to use static relay servers rather than discovering them via DHT by
+using the `libp2p.StaticRelays(relayServerAddrs...)` option on the Host.
+{{% /notice %}}
+
+### Advertising Dialable Addresses
+
+The first section covered how a libp2p nodes “discovers” it’s dialable addresses.
+All the methods and address sets discussed above go into making the final set of a node’s dialable addresses.
+The way a libp2p peer shares these dialable addresses with other peers is:
+
+- When a peer connects to other peers, it asks it’s `Host` for it’s current set of dialable addresses.
+
+- This will include some or all of the address sets discussed above (Interface, Relay, etc. etc.).
```

Comment:
```suggestion
- When a peer connects to other peers, they exchange their dialable addresses via the `Identify` protocol.
    - This will include some or all of the address sets discussed above (Interface, Relay, etc. etc.).
```